### DISCUSSION TITLE
Specify the prisma/schema.prisma file when I execute npx electric-sql generate.

### DISCUSSION THREAD

**Discussion by aquarius-wing** [electric-sql/electric#568] [Feature requests]
Firstly, thank you for creating this fantastic project.

Here's my situation: I have already generated the prisma/schema.prisma file using `npx prisma generate --schema="prisma/schema.prisma"`.

Now, I want to generate `src/generated/client/index.ts`.

When executing `npx electric-sql generate`, I received this log: `npx prisma generate --schema=".electric_migrations_tmp_7uuABl/prisma/schema.prisma"`.

So I got the wrong client which was generated by tmp schema.

Is there action I was doing wrong?

I would be very grateful If you could help me

**Comment by thruflo** [comment]
Hi,

We use prisma tooling as part of our client generation. But we don't run from your Prisma schema. The workflow is:

1. you use Prisma as normal to define your database schema
2. use a Prisma "customised migration" to electrify your tables (currently `CALL electric.electrify(...)`, soon using DDLX statements) as per https://electric-sql.com/docs/integrations/backend/prisma
3. apply your migrations to update your Postgres DDLX
4. *then* run the Electric generator script to generator your client as per https://electric-sql.com/docs/api/generator / https://electric-sql.com/docs/usage/data-access/client

The generator script will pull down the electrified subset of actual your Postgres DDL schema (which you happen to have created/managed using Prisma) and then generate the client from that.

Hope that makes sense!

James

**Reply by aquarius-wing** [reply]
Thanks for reply, the reason why I was doing like this because I already have a sqlite file, and I want make it synchronizable.
Sorry I am new to prisma, like I said, I already have a sqlite file with some data, how to synchronize it?

**Reply by kevin-dp** [reply]
Hi @aquarius-wing, as @thruflo pointed out, our generation script starts from the electrified subset of the Postgres database. So in your case you will need to ensure that your Postgres database reflects the state of your SQLite database. There are many ways how to achieve this. You can manually migrate your Postgres database, or alternatively, you could use a migration tool to help you. For instance, you could use Prisma to introspect your SQLite database, which will result in a Prisma schema, and then you could instruct Prisma to migrate your Postgres database from that schema. The result should be a Postgres database that reflects your SQLite schema. Then, you will need to electrify the tables that you want to replicate to the client. And then you're ready to run the generation script.

**Reply by aquarius-wing** [reply]
Thanks, So, can the image belowe summarizes your instructions?
![image](https://github.com/electric-sql/electric/assets/16633962/8c2f0834-fd9b-44fc-acb3-c31e1e66e00a)

**Reply by kevin-dp** [reply]
Not exactly, you're missing a step.
The steps are:

1. Action: Prisma introspection (cf. [Prisma docs](https://www.prisma.io/docs/concepts/components/introspection))
    Input: SQLite database + initial Prisma schema configured to connect to SQLite DB.
    Output: updated Prisma schema with the models from the SQLite DB
2. Action: Prisma migrate (cf. [Prisma docs](https://www.prisma.io/docs/concepts/components/prisma-migrate))
    Input: Prisma schema (modified to connect to Postgres DB)
    Output: migrated Postgres DB
3. Action: `npx electric-sql generate`
    Output: Electric client

**Reply by aquarius-wing** [reply]
Thank you for your reply.
Now I got this error log `Command failed: npx prisma db pull --schema=".electric_migrations_tmp_R6NOvQ/prisma/schema.prisma`
when i execute command `npx electric-sql generate`

Here are the commands what I run:
1. `npx prisma migrate dev --name init`
2. `npx prisma migrate deploy`

And now here is the structure of database:
![CleanShot 2023-10-18 at 21 01 02@2x](https://github.com/electric-sql/electric/assets/16633962/3c514cda-9b72-4cf1-93f0-1737414737e0)

You can see the table is in there

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
